src/cdk-experimental/scrollable-table-body/scrollable-table-body-layout.ts

```diff
@@ -0,0 +1,111 @@
+ */
+
+import {Directive, Inject, Injectable, Input} from '@angular/core';
+import {CdkTable} from '@angular/cdk/table/table';
+import {DOCUMENT} from '@angular/common';
+import {
+  _TABLE_LAYOUT_STRATEGY,
+  _TableLayout,
+  _TableLayoutStrategy
+} from '@angular/cdk/table/table-layout-strategy';
+
+/**
+ * A {@link _TableLayoutStrategy} that enables scrollable body content for flex tables.
+ */
+@Injectable()
+class ScrollableTableBodyLayoutStrategy implements _TableLayoutStrategy {
+  private _pendingMaxHeight = 'none';
+  private _scrollViewport?: HTMLElement;
+  readonly headerCssClass = 'cdk-table-scrollable-table-header';
+  readonly bodyCssClass = 'cdk-table-scrollable-table-body';
+  readonly footerCssClass = 'cdk-table-scrollable-table-footer';
+
+  constructor(@Inject(DOCUMENT) private readonly _document: any) {
+  }
+
+  /**
+   * Returns the DOM structure for a native table. Scrollable body content is not supported for
+   * native tables. Return `null` to use the default {@link CdkTable} native table layout.
+   */
+  getNativeLayout(table: CdkTable<unknown>): _TableLayout {
+    return null;
+  }
+
+  /**
+   * Returns the DOM structure for a flex table with scrollable body content. Each row outlet
+   * (header, body, footer) is wrapped in a separate container. The specified max height is applied
+   * to the body row outlet to make its content scrollable.
+   */
+  getFlexLayout(table: CdkTable<unknown>): _TableLayout {
+    const documentFragment = this._document.createDocumentFragment();
+    const sections = [
+      {selector: this.headerCssClass, outlets: [table._headerRowOutlet]},
+      {selector: this.bodyCssClass, outlets: [table._rowOutlet, table._noDataRowOutlet]},
+      {selector: this.footerCssClass, outlets: [table._footerRowOutlet]},
+    ];
+
+    for (const section of sections) {
+      const element = this._document.createElement('div');
+      element.classList.add(section.selector);
+      for (const outlet of section.outlets) {
+        element.appendChild(outlet.elementRef.nativeElement);
+      }
+
+      documentFragment.appendChild(element);
+    }
+
+    this._scrollViewport = documentFragment.querySelector(`.${this.bodyCssClass}`);
+    this._scrollViewport!.style.overflow = 'auto';
+    this._applyMaxHeight(this._scrollViewport!, this._pendingMaxHeight);
+
+    return documentFragment;
+  }
+
+  /**
+   * Show a scroll bar if the table's body exceeds this height. The height may be specified with
+   * any valid CSS unit of measurement.
+   */
+  setMaxHeight(v: string) {
+    this._pendingMaxHeight = v;
+    if (this._scrollViewport) {
+      this._applyMaxHeight(this._scrollViewport, v);
+    }
+  }
+
+  private _applyMaxHeight(el: HTMLElement, maxHeight: string) {
+    el.style.maxHeight = maxHeight;
+  }
+}
+
+/** A directive that enables scrollable body content for flex tables. */
+@Directive({
+  selector: 'cdk-table[scrollableBody], mat-table[scrollableBody]',
```

Comment:
It seems weird to be mixing CDK and Material selectors within the same module. Generally we have the CDK as the base class and the Material variant extends it.

Reply:
+1, we would need to separate these